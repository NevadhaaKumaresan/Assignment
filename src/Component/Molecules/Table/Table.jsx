import React from 'react';
import { Table } from 'antd';
import PropTypes from 'prop-types';  


export const TableComponent = (props) => {
    const { taskDatas,columns, action } = props;
    return(
        <Table
        rowKey={record => record.id}
        columns={columns}
        dataSource={action === "pending" ? taskDatas.filter( data => data.completed === false) : 
        taskDatas.filter( data => data.completed === true)}
        pagination={false}
      />
        )
};

TableComponent.propTypes = {
  taskDatas: PropTypes.array.isRequired, 
  columns: PropTypes.array.isRequired, 
  action: PropTypes.string.isRequired, 
}



