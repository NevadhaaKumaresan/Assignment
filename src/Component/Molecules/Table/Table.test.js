/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import { TableComponent } from './Table';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

window.matchMedia = window.matchMedia || function() {
    return {
        matches: false,
        addListener: function() {},
        removeListener: function() {}
    };
  };

test('matches snapshot', () => {
    const taskDatas = [
        {
            "userId": 1,
            "id": 1,
            "title": "delectus aut autem",
            "completed": false
          },
          {
            "userId": 1,
            "id": 2,
            "title": "quis ut nam facilis et officia qui",
            "completed": true
          },
    ]
    const columns = [
        {
            title: 'Title',
            dataIndex: 'title',
          },
          {
            title: 'Complete',
            dataIndex: 'completed',
          },
          {
            title: 'Delete',
            dataIndex: 'deleteIcon',
          }
    ];
    const action="pending";
    const tree = renderer.create(<TableComponent taskDatas={taskDatas} columns={columns} action={action}/>);
    expect(tree).toMatchSnapshot();
  });


  test('should render table component', () => {
    const taskDatas = [
        {
            "userId": 1,
            "id": 1,
            "title": "delectus aut autem",
            "completed": false
          },
          {
            "userId": 1,
            "id": 2,
            "title": "quis ut nam facilis et officia qui",
            "completed": true
          },
    ]
    const columns = [
        {
            title: 'Title',
            dataIndex: 'title',
          },
          {
            title: 'Complete',
            dataIndex: 'completed',
          },
          {
            title: 'Delete',
            dataIndex: 'deleteIcon',
          }
    ];
    const action="completed";
    render(<TableComponent taskDatas={taskDatas} columns={columns} action={action}/>);
  })

  