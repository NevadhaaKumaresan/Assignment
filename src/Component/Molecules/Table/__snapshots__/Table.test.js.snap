// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="ant-table-wrapper"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table"
      >
        <div
          className="ant-table-container"
        >
          <div
            className="ant-table-content"
            onScroll={[Function]}
            style={Object {}}
          >
            <table
              style={
                Object {
                  "tableLayout": "auto",
                }
              }
            >
              <colgroup />
              <thead
                className="ant-table-thead"
              >
                <tr>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Title
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Complete
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Delete
                  </th>
                </tr>
              </thead>
              <tbody
                className="ant-table-tbody"
              >
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key={1}
                  onClick={[Function]}
                  style={Object {}}
                >
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    delectus aut autem
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  />
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  />
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
